server/rpc.go

```diff
@@ -63,19 +68,18 @@ func NewSessionManager(node *core.LivepeerNode, pl core.PlaylistManager) *Broadc
 		sessMap:        make(map[string]*BroadcastSession),
 		createSessions: func() ([]*BroadcastSession, error) { return selectOrchestrator(node, pl) },
 		sessLock:       &sync.Mutex{},
+		minOrchs:       int(HTTPTimeout / SegLen), // maximum inflight sessions
 	}
 	bsm.refreshSessions()
 	return bsm
 }
 
 func (bsm *BroadcastSessionsManager) selectSession() *BroadcastSession {
-	// minOrchs is maximum number of inflight sessions
-	minOrchs := int(HTTPTimeout / SegLen)
 	bsm.sessLock.Lock()
 	defer bsm.sessLock.Unlock()
 	numSess := len(bsm.sessList)
 
-	if numSess < minOrchs {
+	if numSess < bsm.minOrchs {
```

Comment:
This makes testing a bit easier (cleans up some race conditions since refreshSessions will no longer be triggered by default) and also gives us an avenue to handle the case where the session list is naturally shorter than the default minOrchs.